sa/sa_test.go

```diff
@@ -3253,3 +3253,334 @@ func TestGetMaxExpiration(t *testing.T) {
 	test.AssertNotError(t, err, "getting last expriy should succeed")
 	test.Assert(t, lastExpiry.AsTime().Equal(eeCert.NotAfter), "times should be equal")
 }
+
+func TestLeaseOldestCRLShard(t *testing.T) {
+	if os.Getenv("BOULDER_CONFIG_DIR") != "test/config-next" {
+		t.Skip("Test requires crlShards database table")
+	}
+
+	sa, clk, cleanUp := initSA(t)
+	defer cleanUp()
+
+	// Create 8 shards: 4 for each of 2 issuers. For each issuer, one shard is
+	// currently leased, three are available, and one of those failed to update.
+	_, err := sa.dbMap.Exec(
+		`INSERT INTO crlShards (issuerID, idx, thisUpdate, nextUpdate, leasedUntil) VALUES
+		(1, 0, ?, ?, ?),
+		(1, 1, ?, ?, ?),
+		(1, 2, ?, ?, ?),
+		(1, 3, NULL, NULL, ?),
+		(2, 0, ?, ?, ?),
+		(2, 1, ?, ?, ?),
+		(2, 2, ?, ?, ?),
+		(2, 3, NULL, NULL, ?);`,
+		clk.Now().Add(-7*24*time.Hour), clk.Now().Add(3*24*time.Hour), clk.Now().Add(time.Hour),
+		clk.Now().Add(-6*24*time.Hour), clk.Now().Add(4*24*time.Hour), clk.Now().Add(-6*24*time.Hour),
+		clk.Now().Add(-5*24*time.Hour), clk.Now().Add(5*24*time.Hour), clk.Now().Add(-5*24*time.Hour),
+		clk.Now().Add(-4*24*time.Hour),
+		clk.Now().Add(-7*24*time.Hour), clk.Now().Add(3*24*time.Hour), clk.Now().Add(time.Hour),
+		clk.Now().Add(-6*24*time.Hour), clk.Now().Add(4*24*time.Hour), clk.Now().Add(-6*24*time.Hour),
+		clk.Now().Add(-5*24*time.Hour), clk.Now().Add(5*24*time.Hour), clk.Now().Add(-5*24*time.Hour),
+		clk.Now().Add(-4*24*time.Hour),
+	)
+	test.AssertNotError(t, err, "setting up test shards")
+
+	until := time.Now().Add(time.Hour).Truncate(time.Second).UTC()
```

Comment:
This is technically fine, but I think you want:

```suggestion
	until := clk.Now().Add(time.Hour).Truncate(time.Second).UTC()
```